easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,323 @@
+            run_cmd("make yes-%s" % pkg, log_all=True)
+
+        for pkg in self.cfg['packages_no']:
+            self.log.info("Not building %s package", pkg)
+            run_cmd("make no-%s" % pkg, log_all=True)
+
+        # update src/ files with files from pkg subdirs, to propagate changes made by patches (if any)
+        run_cmd("make package-update")
+
+        # put together options for make command
+        cmd_opts = {
+            'build_type': self.cfg['build_type'],
+            'buildopts': self.cfg['buildopts'],
+            'make_vars': self.make_vars,
+            'par': '',
+            'prebuildopts': self.cfg['prebuildopts'],
+        }
+        if self.cfg['parallel']:
+            cmd_opts['par'] = '-j %s' % self.cfg['parallel']
+
+        # run main make commands
+        make_modes = [
+            ('', True),  # LAMMPS itself
+            ('mode=shlib', self.cfg['build_shared_libs']),  # shared libraries
+            ('mode=lib', self.cfg['build_static_libs']),  # static libraries
+        ]
+        for mode, enable in make_modes:
+            if enable:
+                cmd_opts['mode'] = mode
+                cmd = "%(prebuildopts)s make %(par)s %(mode)s %(build_type)s %(make_vars)s %(buildopts)s" % cmd_opts
+                run_cmd(cmd, log_all=True)
+
+    def install_step(self):
+        """Copying files in the right directory"""
+
+        # copy binary and symlink bin/lmp to it
+        lmp_bin = 'lmp_%s' % self.cfg['build_type']
+        bin_dir = os.path.join(self.installdir, 'bin')
+        copy_file(os.path.join(self.start_dir, 'src', lmp_bin), os.path.join(bin_dir, lmp_bin))
+        symlink(os.path.join(bin_dir, lmp_bin), os.path.join(bin_dir, 'lmp'))
+
+        # copy libraries to standard location <prefix>/lib
+        for lib in glob.glob('liblammps*'):
+            copy_file(lib, os.path.join(self.installdir, 'lib', lib))
+
+        # install Python bindings
+        if self.lammps_pylibdir:
+            change_dir(os.path.join(self.start_dir, 'python'))
+            pylibdir = os.path.join(self.installdir, self.lammps_pylibdir)
+            mkdir(pylibdir, parents=True)
+            cmd = "python install.py %s" % pylibdir
+            run_cmd(cmd, log_all=True)
+
+    def sanity_check_step(self):
+        """Custom sanity check for LAMMPS."""
+
+        build_type = self.cfg['build_type']
+
+        files = ['bin/lmp', 'bin/lmp_%s' % build_type]
+        custom_commands = []
+
+        if self.cfg['build_shared_libs']:
+            shlib_ext = get_shared_lib_ext()
+            files.extend(['lib/liblammps.%s' % shlib_ext, 'lib/liblammps_%s.%s' % (build_type, shlib_ext)])
+
+        if self.cfg['build_static_libs']:
+            files.extend(['lib/liblammps.a', 'lib/liblammps_%s.a' % build_type])
+
+        if self.lammps_pylibdir:
+            files.append(os.path.join(self.lammps_pylibdir, 'lammps.py'))
+            custom_commands.append("python -c 'from lammps import lammps; lmp = lammps()'")
```

Comment:
For what it's worth, if I add `sanity_check_commands = ['']` to the easyconfig, it installs fine, and then if I load the module, the sanity check command runs just fine,
```[mboisson@build-node easyconfigs]$ python -c "from lammps import lammps; lmp = lammps()"
LAMMPS (11 Aug 2017)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (../comm.cpp:90)
  using 1 OpenMP thread(s) per MPI task
Total wall time: 0:00:00```

Reply:
Ok, this is partly linked to https://github.com/easybuilders/easybuild-framework/issues/2499

However, this will still fail if python is a build dependency rather than a runtime dependency.